src/jit/compiler.h

```diff
@@ -8297,6 +8277,11 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
         }
 #endif
 
+        bool IsPreJIT()
```

Comment:
Why do you need to check whether this is PreJIT?

This should be based solely on the available instruction set, no?

Reply:
Ok, I see.

Would it be better to filter out the problematic methods in `FilterHardwareIntrinsicMethodAttribs` instead. I think it would be nice to have the hacks required to make the intrinsics in CoreLib work in a central place.

Reply:
I went with this as I believed it to be the minimal set of changes needed for netcoreapp3.0 and would have the least impact on what code could be AOT'd.

If we are just talking about S.P.Corelib, I *think* we would only need to filter out the `Math.Round`, `Math.Ceiling`, and `Math.Floor` methods (as the other relevant changes apply to the `Vector64/128/256<T>` helper methods, which should already be getting filtered).

If we go a bit broader, this also impacts various codegen around `FEATURE_SIMD` with relation to the "SIMD support level" which can be `AVX2`, `SSE41`, or `SSE2`. At first glance, that mostly looks limited to the SIMD intrinsics (and therefore, outside of `Vector<T>`, to the `System.Numerics` library), but there is quite a bit of handling around those and I didn't go through it all.

Reply:
If we did filter them in `FilterHardwareIntrinsicMethodAttribs`, that might also limit the AOT codegen quality if/when we expand the support beyond S.P.Corelib (but that could also be handled at that point in time.)

Reply:
The strategy to expand to support outside CoreLib will need to be different. So I think it would be best to keep the hack required to make the CoreLib work centralized, and not spread it around.

Reply:
Hmmm, `Math.Ceiling` and `Math.Floor` might be problematic. They are still FCALLs and so only get `CORINFO_FLG_INTRINSIC`. `Math.Round`, however, is a named intrinsic and so can go through the normal filtering.

Do we have a good way of allowing `Math.Ceiling`/`Math.Floor` to still be resolved to their CRT helper but not allowing the additional intrinsic handling?

Reply:
In `ZapInfo::getIntrinsicID`, check the `CORINFO_INTRINSIC_Ceiling` / `CORINFO_INTRINSIC_Floor` and return `CORINFO_INTRINSIC_Illegal` instead.